packages/aws-cdk-lib/aws-cloudfront-origins/test/function-url-origin.test.ts

```diff
@@ -41,3 +44,174 @@ test('Correctly renders the origin for a Lambda Function URL', () => {
     },
   });
 });
+
+test('Correctly sets readTimeout and keepaliveTimeout', () => {
+  const fn = new lambda.Function(stack, 'MyFunction', {
+    code: lambda.Code.fromInline('exports.handler = async () => {};'),
+    handler: 'index.handler',
+    runtime: lambda.Runtime.NODEJS_20_X,
+  });
+
+  const fnUrl = fn.addFunctionUrl({
+    authType: lambda.FunctionUrlAuthType.NONE,
+  });
+
+  const origin = new FunctionUrlOrigin(fnUrl, {
+    readTimeout: cdk.Duration.seconds(120),
+    keepaliveTimeout: cdk.Duration.seconds(60),
+  });
+
+  const originBindConfig = origin.bind(stack, { originId: 'StackOriginLambdaFunctionURL' });
+
+  expect(stack.resolve(originBindConfig.originProperty)).toMatchObject({
+    customOriginConfig: {
+      originReadTimeout: 120,
+      originKeepaliveTimeout: 60,
+    },
+  });
+});
+
+test('Correctly adds permission to Lambda for CloudFront', () => {
+  const fn = new lambda.Function(stack, 'MyFunction', {
+    code: lambda.Code.fromInline('exports.handler = async () => {};'),
+    handler: 'index.handler',
+    runtime: lambda.Runtime.NODEJS_20_X,
+  });
+
+  const fnUrl = fn.addFunctionUrl({
+    authType: lambda.FunctionUrlAuthType.AWS_IAM,
+  });
+
+  new cloudfront.Distribution(stack, 'MyDistribution', {
+    defaultBehavior: {
+      origin: FunctionUrlOrigin.withOriginAccessControl(fnUrl, {
+        originAccessControl: undefined,
+      }),
+    },
+  });
+
+  const template = Template.fromStack(stack);
+
+  template.hasResourceProperties('AWS::CloudFront::Distribution', {
```

Comment:
since this test is checking for the correct permissions, can we also check the `AWS::Lambda::Permission` resource exists in the template and has the expected policy?

Reply:
fixed.
https://github.com/aws/aws-cdk/pull/31339/commits/7226680a2e61894796626e5e23b54f5d9586490d